### ISSUE TITLE
tcp_if_seq not defined in mca_btl_tcp_component_t

### ISSUE THREAD

**Issue by amckinstry** [open-mpi/ompi#2891]
Good morning, and congratulations on releasing 2.0.2.
Now for the bad news :-(

Compilation for me fails in btl_tcp_component.c as:
```
#if !defined(MCA_BTL_TCP_SUPPORT_PROGRESS_THREAD)
    if( mca_btl_tcp_component.tcp_enable_progress_thread ) {
        opal_show_help("help-mpi-btl-tcp.txt",
                       "unsuported progress thread",
                       true, "progress thread",
                       opal_process_info.nodename,
                       mca_btl_tcp_component.tcp_if_seq,
                       "Progress thread support compiled out");
     }
#endif  /* !defined(MCA_BTL_TCP_SUPPORT_PROGRESS_THREAD) */
```
tcp_if_seq not defined.

**Comment by amckinstry** [comment]
This appears to be fixed in HEAD with commit a49422fe843eec9b4d3676b30015483dbf4ebc3c

**Comment by jsquyres** [comment]
Can you send us your exact configure command line?

**Comment by amckinstry** [comment]
./configure --build=x86_64-linux-gnu --prefix=/usr --includedir=${prefix}/include --mandir=${prefix}/share/man --infodir=${prefix}/share/info --sysconfdir=/etc --localstatedir=/var --disable-silent-rules --libdir=${prefix}/lib/x86_64-linux-gnu --libexecdir=${prefix}/lib/x86_64-linux-gnu --disable-maintainer-mode --disable-dependency-tracking --with-libfabric --with-psm --with-jdk-dir=/usr/lib/jvm/default-java --enable-mpi-java --enable-opal-btl-usnic-unit-tests --disable-wrapper-rpath --enable-mpi-thread-multiple --disable-silent-rules --enable-mpi-cxx --with-hwloc=/usr/ --with-libltdl=/usr/ --with-devel-headers --with-slurm --with-sge --without-tm --enable-heterogeneous --disable-vt --sysconfdir=/etc/openmpi --libdir=${prefix}/lib/x86_64-linux-gnu/openmpi/lib --includedir=${prefix}/lib/x86_64-linux-gnu/openmpi/include

**Comment by amckinstry** [comment]
Applying the commit, I get a second failure:
```
../../../opal/.libs/libopen-pal.so: undefined reference to `mca_base_var_enum_create_flag'
collect2: error: ld returned 1 exit status
Makefile:1901: recipe for target 'opal_wrapper' failed
```
which is indeed the case. Its used, defined in the header, but missing from mca_base_var_enum.c.

**Comment by ggouaillardet** [comment]
@amckinstry i was unable to reproduce this on an up-to-date CentOS 7

btw, which distro are you running on ?

at first glance,
1. it should be
```v
#if ! MCA_BTL_TCP_SUPPORT_PROGRESS_THREAD
```
instead

2. that being said, from btl_tcp.h
```
#if (HAVE_PTHREAD_H == 1)
#define MCA_BTL_TCP_SUPPORT_PROGRESS_THREAD 1
#else
#define MCA_BTL_TCP_SUPPORT_PROGRESS_THREAD 0
#endif  /* (HAVE_PTHREAD_H == 1) */
```

it should be
```c
#ifdef HAVE_PTHREAD_H
```

and that also being said, the ```MCA_BTL_TCP_SUPPORT_PROGRESS_THREAD``` macro is always set,
so there should be no errors

also, i was unable to find any ```mca_base_var_enum_create_flag``` in the openmpi-2.0.2 tarball

**Comment by rhc54** [comment]
@amckinstry I confirmed what @ggouaillardet said - there is no ```mca_base_var_enum_create_flag``` in the 2.0 series. I suspect you have a mixed code tree that overlays 2.0.2 with some other version of OMPI. You might want to delete it and start with a fresh unpack of a 2.0.2 tarball

**Comment by amckinstry** [comment]
@rhc54 Re-checking with a clean 2.0.2 tarball, I agree, `mca_base_var_enum_create_flag' is a local failure on my part, due to a corrupted merge.

@ggouaillardet I'm the Debian (/Ubuntu) package mantainer :-)

When I backed out a local patch I had to fix the MCA_BTL_TCP_SUPPORT_PROGRESS_THREAD issue (and used a clean checkout), it still compiles.

So, I can no longer trigger the issue on a clean compile (HAVE_PTHREAD_H is always defined), but the tcp_if_seq reference is a latent bug, as its not defined in the rest of the code.
